aten/src/ATen/native/Cross.cpp

```diff
@@ -1,52 +1,71 @@
 #include <ATen/ATen.h>
 #include <ATen/Dispatch.h>
 #include <ATen/NativeFunctions.h>
+#include <ATen/native/Resize.h>
+#include <ATen/ExpandUtils.h>
 
 #include <ATen/native/Cross.h>
 
 namespace at { namespace native {
 
 DEFINE_DISPATCH(cross_stub);
 
+int64_t _default_cross_dim(const c10::optional<int64_t> &dimension, IntArrayRef sizes) {
+  if(dimension.has_value()) {
+    return *dimension;
+  }
+
+  for(auto i : c10::irange(sizes.size())) {
+    if(sizes[i] == 3) {
+      return i;
+    }
+  }
+  TORCH_CHECK(false, "no dimension of size 3 in input");
+}
+
 Tensor cross(const Tensor & input, const Tensor & other, const c10::optional<int64_t> dimension) {
-  Tensor out = at::empty_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
-  native::cross_out(input, other, dimension, out);
-  return out;
+  auto dim = _default_cross_dim(dimension, input.sizes());
+  return at::linalg_cross(input, other, dim);
 }
 
 Tensor & cross_out(const Tensor & input, const Tensor & other, const c10::optional<int64_t> dimension, Tensor & out) {
-  auto device_res = input.device().type();
-  TORCH_CHECK(device_res == kCPU || device_res == kCUDA, "cross only supports CPU and CUDA devices, out got: ", device_res);
+  auto dim = _default_cross_dim(dimension, input.sizes());
+  return at::linalg_cross_out(out, input, other, dim);
+}
+
+Tensor linalg_cross(const Tensor & input, const Tensor & other, const int64_t dimension) {
+  Tensor out = at::empty({0}, input.options());
+  native::linalg_cross_out(input, other, dimension, out);
+  return out;
+}
+
+Tensor & linalg_cross_out(const Tensor & input, const Tensor & other, const int64_t dimension, Tensor & out) {
   auto device1 = input.device().type();
```

Comment:
Nit. This variable can be deleted and you can call `stub` directly with `input.device().type()`.

Reply:
Yeah, nice cleanup. Done!